### ISSUE TITLE
Maintenance Mode in app_dev.php

### ISSUE THREAD

**Issue by christianromeni** [contao/core-bundle#527]
When activating the maintenance mode in 4.2, there is an error 503 - Service unavailable when using app_dev.php when loged out of the backend.
It would be awesome if the app_dev.php could continue working even if maintenance mode is enabled, so it is possible to show the page in the dev envirdonment without loging in twice (frontend & backend)

![image](https://cloud.githubusercontent.com/assets/1358040/16317538/e31d6bf0-398b-11e6-827e-d08a840a47cf.png)

![image](https://cloud.githubusercontent.com/assets/1358040/16317531/d9922c60-398b-11e6-8ad7-b6442ccf840a.png)

**Comment by leofeyer** [comment]
This seems reasonable to me. @contao/developers Any objections?

**Comment by aschempp** [comment]
Fine with me :-)

**Comment by Toflar** [comment]
Okay for me 👍

**Comment by leofeyer** [comment]
I have moved the ticket to the next minor release, because we will have to change the constructor arguments of the `BypassMaintenanceListener`:

```php
// Before
public function __construct(SessionInterface $session, $disableIpCheck, $requestAttribute = '_bypass_maintenance')

// After
public function __construct(SessionInterface $session, $disableIpCheck, $debug, $requestAttribute = '_bypass_maintenance')
```

Since there is no BC promise for event listeners, I guess the change can go into the next minor version, right?

**Comment by aschempp** [comment]
It could be in any version if we add `$debug` at the end and provide a sane default value…

**Comment by leofeyer** [comment]
Me not like.

**Comment by discordier** [comment]
I am with @aschempp on this one, we should add it to the end and default to false.

**Comment by leofeyer** [comment]
There is a much simpler solution, which I have implemented in contao/standard-edition@7696306a44469874d02c68d544a8521d076b5dcb and which everyone can use immediately by adjusting their app configuration.
